lib/tasks/gitlab/check.rake

```diff
@@ -278,6 +279,29 @@ namespace :gitlab do
       end
     end
 
+    def check_upload_perms
+      print "Upload directory permissions correct? ... "
+
+      upload_path = Rails.root.join(Gitlab.config.gitlab.user_home,"gitlab-rails","uploads")
+      upload_path_tmp = Rails.root.join(upload_path,"tmp")
```

Comment:
Thanks for asking @bbodenmiller . I think I was a little rushed above :)

I think the intended use of 'Rails.root.join' is to find the path to files/directories under either `/home/git/gitlab` or `/opt/gitlab/service/embedded/gitlab-rails`. GitLab the Rails app expects the uploads to be in `public/uploads` under the Rails root, hence `Rails.root.join('public/uploads')`. Now we know that with the omnibus packages, `public/uploads` is actually a symlink. By using `File.realpath` we can find out what it points to.

The gain of all this is that we are no longer making assumptions about the file layout beyond 'uploads are in `public/uploads`'.